students/ChrisH/lesson03/locke_operation_model.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+# ----------------------------------------------------------------
+# Contains a context manager class that simulates the overall
+# functioning of a waterway locke system. (Convention - spelled
+# locke when used as a noun. Spelled lock when verb.)
+# System functions:
+#   when locke entered, stops pumps, opens doors, closes doors, restarts pumps
+# When initialized, accepts capacity in number of boats. Raises error
+# if too many boats try to move through Locke.
+# ----------------------------------------------------------------
+
+
+class Locke(object):
+    """
+    Context Manager class that simulates functioning of a waterway locke.
+    :param capacity: number of boats locke can take
+    """
+
+    def __init__(self, capacity):
+        if capacity < 1:
+            raise ValueError('Capacity must be greater than zero.')
+        self.capacity = capacity
+
+    def __enter__(self):
+        print("Stopping the pumps.")
+        print("Opening the doors.")
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        print("Closing the doors.")
+        print("Restarting the pumps.")
+        return self
```

Comment:
why are you returning self?

Reply:
Ah, well, I'm going to go ahead and say I'm not exactly sure why. 
Ah, it turns out that I need to return self in the \_\_enter\_\_ function if I'm instantiating the object in the with statement. Thus, providing an object that disappears when the "with" block is done.
```
with Locke(10) as locke_10:
    print(locke_10.capacity)
```
But... yeah, I can't see why I would need it in the \_\_exit\_\_ function. Because, it would be returned to... huh, who knows where?